gensim/models/word2vec_inner.pyx

```diff
@@ -343,6 +343,107 @@ def train_sentence_sg(model, sentence, alpha, _work):
     return result
 
 
+def train_batch_sg(model, sentences, alpha, sentence_indeces, _work):
+    cdef int hs = model.hs
+    cdef int negative = model.negative
+    cdef int sample = (model.sample != 0)
+
+    cdef REAL_t *syn0 = <REAL_t *>(np.PyArray_DATA(model.syn0))
+    cdef REAL_t *word_locks = <REAL_t *>(np.PyArray_DATA(model.syn0_lockf))
+    cdef REAL_t *work
+    cdef REAL_t _alpha = alpha
+    cdef int size = model.layer1_size
+
+    cdef int codelens[MAX_SENTENCE_LEN]
+    cdef np.uint32_t indexes[MAX_SENTENCE_LEN]
+    cdef np.uint32_t reduced_windows[MAX_SENTENCE_LEN]
+    cdef int sentence_len[MAX_NUM_SENTENCES]
+    cdef int sentence_indeces_c[MAX_NUM_SENTENCES]
```

Comment:
I don't believe 'indeces' is a typical pluralization of 'index' anywhere. FWIW, the use of 'indexes' vs 'indices' was mixed in this code (especially between the pure-python and cython files) when I first started optimizing/extending it. There's a reasonable case to be made for either spelling, but I prefer 'indexes' so that greps for 'index' also find related lists/arrays.